apps/jobs/views.py

```diff
@@ -688,23 +689,39 @@ def get_submission_by_pk(request, submission_id):
         return Response(response_data, status=status.HTTP_404_NOT_FOUND)
 
     host_team = submission.challenge_phase.challenge.creator
+    if request.method == "GET":
+        if (
+            request.user.id == submission.created_by.id
+            or ChallengeHost.objects.filter(
+                user=request.user.id, team_name__pk=host_team.pk
+            ).exists()
+        ):
+            serializer = SubmissionSerializer(
+                submission, context={"request": request}
+            )
+            response_data = serializer.data
+            return Response(response_data, status=status.HTTP_200_OK)
 
-    if (
-        request.user.id == submission.created_by.id
-        or ChallengeHost.objects.filter(
-            user=request.user.id, team_name__pk=host_team.pk
-        ).exists()
-    ):
+        response_data = {
+            "error": "Sorry, you are not authorized to access this submission."
+        }
+        return Response(response_data, status=status.HTTP_401_UNAUTHORIZED)
+
+    elif request.method == "DELETE":
```

Comment:
Hey @Kajol-Kumari, This `DELETE` API should only be accessible to challenge host of a challenge and not anyone else. So, this option should appear in "All Submissions" and API should have a check for the user as a host otherwise it should reject the request.

Reply:
Added the check for challenge host in the API and the delete option is provided under `All submission` tab. Please look at it @RishabhJain2018.